### PR TITLE
Start codegen

### PR DESCRIPTION
**Pull request by JakeRoggenbuck**

### CODE REVIEW

**Comment by TheCrystalKeeper** [comment]
Seems fine to me -- doesn't look like there are any changes that break anything.

File: src/codegen/x86/codegen.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+mov r12, [rsp]\
+mov r13, [rsp+8]\
+mov r14, [rsp+16]\
+mov r15, [rsp+24]\
+add rsp, 32";
+
+    return end;
+}
+
+char *init_int_literal(int val) {
```
Comment (adamhutchings) [inline src/codegen/x86/codegen.c:63]: For my understanding, what exactly does this do? Does it just push a new int with the specified value to the stack? Still getting there with the assembly.

Also, this is all fine for now, but in the future maybe we should have all of these functions also accept a `FILE* fp` and print into there instead. What do you think?

Comment (JakeRoggenbuck) [reply src/codegen/x86/codegen.c:63]: Yup

File: src/codegen/x86/codegen.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+char *end_func() {
+    static char end[256] = "\
+mov r12, [rsp]\
+mov r13, [rsp+8]\
+mov r14, [rsp+16]\
+mov r15, [rsp+24]\
+add rsp, 32";
+
+    return end;
+}
```
Comment (NicoBliss) [inline src/codegen/x86/codegen.c:61]: This is perhaps silly, but are all of these no-input functions not behaving as immutable global variables you are asking for copies of?

Comment (adamhutchings) [reply src/codegen/x86/codegen.c:61]: Given the conversation above, it will make sense once we reorganize this to write into files instead

**Pull request merged**